### PR TITLE
Fix amount not getting passed through in simple contribution flow

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fixes regression in 5.69 whereby `amount` was not passed through correctly in some flows. This was masked by some of the tests using an old flow which had a hack in it that made it appear to work. I have removed that hack from the tests & fixed the pass through

@totten confirmed it worked in his r-run

Before
----------------------------------------
`amount` not passed through to the payment processor in fixed flows, test coverage hampered by test hack

After
----------------------------------------
hack removed

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/Form/Contribution/Confirm.php [hunk 1]
```diff
@@ -1863,7 +1864,6 @@ public static function submit($params) {
     if ($form->isSeparateMembershipPayment()) {
       $amount -= $form->order->getMembershipTotalAmount();
     }
-    $form->_amount = $params['amount'] = $form->_params['amount'] = $amount;
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Contribution/Confirm.php:1866]: this line is only called by tests - it was creating false passes

**Comment by seamuslee001** [comment]
@eileenmcnaughton test fail looks related

```
api_v3_ContributionTest::testPayLater
' contains "amount:::100".

/home/homer/buildkit/build/build-2/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviMailUtils.php:285
/home/homer/buildkit/build/build-2/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviMailUtils.php:173
/home/homer/buildkit/build/build-2/web/sites/all/modules/civicrm/tests/phpunit/api/v3/ContributionTest.php:2086
/home/homer/buildkit/build/build-2/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviUnitTestCase.php:247
/home/homer/buildkit/extern/phpunit9/phpunit9.phar:2307
```

**Comment by totten** [comment]
`r-run`: I've been comparing 5.68/5.69/this patch (circa 42e06f919ff00ebed6c50864b7954b2e902838a2 or f726c26943e6013ae411ab9ff63265b0006ef900) and running Stripe/iATS-CC in test mode. 

The patch fixes a recent regression (within 5.69) where new contributions are stored with a value of `$0`.  Huzzah

**Comment by eileenmcnaughton** [comment]
thanks @totten @seamuslee001  - the test fail was a variable that is no longer in the template so I updated the test to test the token instead

**Comment by totten** [comment]
@eileenmcnaughton Looks like we still have test failures

**Comment by eileenmcnaughton** [comment]
thanks @colemanw

**Pull request merged**